alpenhorn/util.py

```diff
@@ -15,36 +15,51 @@
 log = logging.getLogger(__name__)
 
 
-def run_command(cmd, **kwargs):
+def run_command(
+    cmd: list[str], timeout: float | None = None, **kwargs
+) -> tuple(int | None, str, str):
     """Run a command.
 
     Parameters
     ----------
-    cmd : array
+    cmd : list of strings
         A command as a list of strings including all arguments.
-    kwargs : dict
-        Passed directly onto `subprocess.Popen.`
+    timeout : float or None
+        Number of seconds to wait before forceably killing the process,
+        or None to wait forever.
+
+    Other keyword args are passed directly on to subprocess.Popen
 
     Returns
     -------
-    retval : int
-        Return code.
-    stdout_val : string
-        Value of stdout.
-    stderr_val : string
-        Value of stderr.
+    Returns a three-element tuple containing:
+        retval : int or None
+            Return code, or None if the process was killed after timing out.
+        stdout : string
+            Value of stdout.
+        stderr : string
+            Value of stderr.
```

Comment:
```suggestion
    retval : int or None
        Return code, or None if the process was killed after timing out.
    stdout : string
        Value of stdout.
    stderr : string
        Value of stderr.
```
Since python packages multiple returns as a tuple by default, I think we can just list all the returns without having to specify that they're in a tuple (especially now with the extra type hinting)